package/kedro_viz/launchers/cli.py

```diff
@@ -87,10 +90,51 @@ def commands():
     help="Kedro configuration environment. If not specified, "
     "catalog config in `local` will be used",
 )
-def viz(host, port, browser, load_file, save_file, pipeline, env):
+@click.option(
+    "--autoreload",
+    is_flag=True,
+    help="Autoreload viz server when a Python file change in the Kedro project",
+)
+def viz(host, port, browser, load_file, save_file, pipeline, env, autoreload):
     """Visualise a Kedro pipeline using Kedro viz."""
     try:
-        run_server(host, port, browser, load_file, save_file, pipeline, env)
+        if autoreload:
+            is_localhost = host in ("127.0.0.1", "localhost", "0.0.0.0")
+
+            if browser and is_localhost:
+                webbrowser.open_new(f"http://{host}:{port}/")
+
+            project_path = Path.cwd()
+            process_kwargs = {
+                "host": host,
+                "port": port,
+                "load_file": load_file,
+                "save_file": save_file,
+                "pipeline_name": pipeline,
+                "env": env,
+                "autoreload": True,
+                "browser": False,
+                "project_path": project_path,
+            }
```

Comment:
These look very similar to the arguments passed directly to `run_server()` in the `else` branch. Reusing them, with `"autoreload": autoreload, "browser": browser`, across both calls might make it easier to follow what the difference is (which as I understand it is just that we pass a `project_path` if we're in reload mode?